### ISSUE TITLE
crash: wallet.get_full_history - self.txo[txid] KeyError

### ISSUE THREAD

**Issue by SomberNight** [spesmilo/electrum#3938]
On git master d77e522721b98c8d455f866bc40b778c6a0fb74e.

```
Traceback (most recent call last):
  File "/home/user/wspace/electrum/gui/qt/__init__.py", line 217, in start_new_window
    w = self.create_window_for_wallet(wallet)
  File "/home/user/wspace/electrum/gui/qt/__init__.py", line 179, in create_window_for_wallet
    w = ElectrumWindow(self, wallet)
  File "/home/user/wspace/electrum/gui/qt/main_window.py", line 201, in __init__
    self.load_wallet(wallet)
  File "/home/user/wspace/electrum/lib/util.py", line 275, in <lambda>
    return lambda *args, **kw_args: do_profile(func, args, kw_args)
  File "/home/user/wspace/electrum/lib/util.py", line 271, in do_profile
    o = func(*args, **kw_args)
  File "/home/user/wspace/electrum/gui/qt/main_window.py", line 344, in load_wallet
    self.history_list.update()
  File "/home/user/wspace/electrum/gui/qt/util.py", line 484, in update
    self.on_update()
  File "/home/user/wspace/electrum/lib/util.py", line 275, in <lambda>
    return lambda *args, **kw_args: do_profile(func, args, kw_args)
  File "/home/user/wspace/electrum/lib/util.py", line 271, in do_profile
    o = func(*args, **kw_args)
  File "/home/user/wspace/electrum/gui/qt/history_list.py", line 202, in on_update
    r = self.wallet.get_full_history(domain=self.get_domain(), from_timestamp=self.start_timestamp, to_timestamp=self.end_timestamp, fx=fx)
  File "/home/user/wspace/electrum/lib/util.py", line 275, in <lambda>
    return lambda *args, **kw_args: do_profile(func, args, kw_args)
  File "/home/user/wspace/electrum/lib/util.py", line 271, in do_profile
    o = func(*args, **kw_args)
  File "/home/user/wspace/electrum/lib/wallet.py", line 1019, in get_full_history
    ap, lp = self.capital_gain(tx_hash, fx.timestamp_rate, fx.ccy)
  File "/home/user/wspace/electrum/lib/wallet.py", line 1709, in capital_gain
    acquisition_price = out_value/Decimal(COIN) * self.average_price(tx, price_func, ccy)
  File "/home/user/wspace/electrum/lib/wallet.py", line 1714, in average_price
    input_value = sum(self.txin_value(txin) for txin in tx.inputs()) / Decimal(COIN)
  File "/home/user/wspace/electrum/lib/wallet.py", line 1714, in <genexpr>
    input_value = sum(self.txin_value(txin) for txin in tx.inputs()) / Decimal(COIN)
  File "/home/user/wspace/electrum/lib/wallet.py", line 1675, in txin_value
    for address, d in self.txo[txid].items():
KeyError: '3a20f720b0b5f6d042d757a7c1d0c474d71c976efca23566328a8825f11f2032'
```

**Comment by ecdsa** [comment]
fixed in 51f04d4e7bdb1021c5535c47b676c3d2d084eebf

**Comment by SomberNight** [comment]
I looked into the underlying cause.

Do capital gains calc on all txns in history:
https://github.com/spesmilo/electrum/blob/d77e522721b98c8d455f866bc40b778c6a0fb74e/lib/wallet.py#L1019

Calc avg price of input coins for tx:
https://github.com/spesmilo/electrum/blob/d77e522721b98c8d455f866bc40b778c6a0fb74e/lib/wallet.py#L1709

Get value of inputs from outputs of parent txns:
https://github.com/spesmilo/electrum/blob/d77e522721b98c8d455f866bc40b778c6a0fb74e/lib/wallet.py#L1714
https://github.com/spesmilo/electrum/blob/d77e522721b98c8d455f866bc40b778c6a0fb74e/lib/wallet.py#L1672-L1675

But if the original txn is an incoming txn, we won't have the parent txn.
